pkg/steps/cluster_claim_test.go

```diff
@@ -0,0 +1,231 @@
+package steps
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	prowv1 "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/pod-utils/downwardapi"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	fakectrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/steps/loggingclient"
+	"github.com/openshift/ci-tools/pkg/testhelper"
+	fakewatchingclient "github.com/openshift/ci-tools/pkg/util/watchingclient/fake"
+)
+
+var (
+	aClusterPool = &hivev1.ClusterPool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "ci-ocp-4.7.0-amd64-aws-us-east-1",
+			Namespace: "ci-cluster-pool",
+			Labels: map[string]string{
+				"product":      "ocp",
+				"version":      "4.7.0",
+				"architecture": "amd64",
+				"cloud":        "aws",
+				"owner":        "dpp",
+				"region":       "us-east-1",
+			},
+		},
+	}
+	aClusterDeployment = &hivev1.ClusterDeployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "ci-ocp-4.7.0-amd64-aws-us-east-1-ccx23",
+			Namespace: "ci-ocp-4.7.0-amd64-aws-us-east-1-ccx23",
+		},
+		Spec: hivev1.ClusterDeploymentSpec{
+			ClusterMetadata: &hivev1.ClusterMetadata{
+				AdminKubeconfigSecretRef: corev1.LocalObjectReference{
+					Name: "ci-openshift-46-aws-us-east-1-ccx23-0-gpjsf-admin-kubeconfig",
+				},
+				AdminPasswordSecretRef: corev1.LocalObjectReference{
+					Name: "ci-openshift-46-aws-us-east-1-ccx23-0-gpjsf-admin-password",
+				},
+			},
+		},
+	}
+	aKubeconfigSecret = corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "ci-openshift-46-aws-us-east-1-ccx23-0-gpjsf-admin-kubeconfig",
+			Namespace: "ci-ocp-4.7.0-amd64-aws-us-east-1-ccx23",
+		},
+		Data: map[string][]byte{
+			"kubeconfig": []byte("some-kubeconfig"),
+		},
+	}
+	aPasswordSecret = corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "ci-openshift-46-aws-us-east-1-ccx23-0-gpjsf-admin-password",
+			Namespace: "ci-ocp-4.7.0-amd64-aws-us-east-1-ccx23",
+		},
+		Data: map[string][]byte{
+			"password": []byte("some-kubeadmin-password"),
+		},
+	}
+)
+
+func TestClusterClaimStepRun(t *testing.T) {
+	if err := hivev1.AddToScheme(scheme.Scheme); err != nil {
```

Comment:
this is not threadsafe, could you move it to an `init`? Otherwise it might race if multiple tests do this

Reply:
Updated.